client/src/cli/commands/logs.py

```diff
@@ -0,0 +1,37 @@
+import argparse
+from pprint import pp
+from typing import Any
+
+import openapi_client
+
+from .util import with_job_mgmt_api
+
+
+@with_job_mgmt_api
+def logs(client: openapi_client.JobManagementApi, args: argparse.Namespace) -> None:
+    params = {k: v for k, v in vars(args).items() if v is not None}
+
+    resp = client.logs_jobs_uid_logs_get(**params)
+    pp(resp)
```

Comment:
This does not deal with the streaming mode properly, where the response is not closed until the pod terminates.

We'll need an async approach for this.

Reply:
Correct me if I'm wrong, but it seems that `kubectl logs <pod> -f` also creates a blocking log stream for continuously reading from a pod. Not to mention that to get already written logs (described [here](https://kubernetes.io/docs/concepts/cluster-administration/logging/#log-rotation)), we can use a synchronous call.

What is the use case for asynchronously requesting streaming logs? In my understanding, streaming communicates the desire to be at the "pulse" of a running pod, which means that I want to read the logs themselves immediately or at least very shortly after streaming. In that case, I am probably fine with using a blocking CLI call, since my full attention is on inspecting the incoming logs. Couldn't we just use a HTTP streaming response and avoid async?

Plus, to redirect streamed logs into a file, we can use `tee` and keep it entirely in the shell like you suggested elsewhere in the review ([link](https://github.com/aai-institute/infrastructure-product/pull/65#discussion_r1738030100)):

```console
jobby logs <my-workload> --stream | tee my-workload-logs.txt
```

This works fine when tested on another streaming application like good old `ping 8.8.8.8`. That together with tail logging seems to support the whole available pod-level k8s logs API, if I am not mistaken. Could you clarify the async CLI workflow you mentioned?

Reply:
Addendum: We tested without the httpx branch on a previous commit, and found that `jobby logs` blocks until the entire pod logs are available in the backend for the streaming response, at which point the streaming is useless. So the async facility is actually necessary.

Reply:
See below, applied a solution in 8f9a6a8fafe321e660690f8e88f613c59501823f.